common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/ImageIndexableTest.scala

```diff
@@ -143,19 +143,34 @@ class ImageIndexableTest
       }
     }
 
-    it("throws an error if an image has >= 10 merged sources") {
-      val erroneousImage = createAugmentedImage().copy(
+    it("clamps the version if an image has >= versionMultiplier sources") {
+      val nSources = Indexable.versionMultiplier + 10
+      val lotsOfSourcesImage = createAugmentedImage().copy(
         source = SourceWorks(
           canonicalWork = identifiedWork().toSourceWork,
           redirectedWork = None,
-          nMergedSources = 10
+          nSources = nSources
         )
       )
+      val slightlyFewerSourcesImage = lotsOfSourcesImage.copy(
+        source = lotsOfSourcesImage.source
+          .asInstanceOf[SourceWorks[DataState.Identified]]
+          .copy(nSources = nSources - 1)
+      )
```

Comment:
This test doesn't feel right to me – you're testing a side-effect of the implementation, rather than the behaviour we want.

What we want:

> If I ingest V1 with lots of sources, V2 will still win out no matter how many sources it has.

I think you want to bump the version and drop `nSources` to the minimum (1?) for a more meaningful test.

Reply:
You are describing a separate behaviour to what this is testing - as we clamp _at_ the maximum, this is testing that above the maximum number of sources, even a work with fewer (but still too many) sources will override an existing one. I think this is important to document in the test, so I will add another test for the behaviour you're describing.